2D Simulation/game.js

```diff
@@ -154,33 +158,38 @@ function update_gui_block(moveData) {
 }
 
 socket.on('update_flip_block', function (block_id) {
-    let id = block_id.substring(5);
+    update_flip_block(block_id);
+});
 
+function update_flip_block(block_id) {
+    let id = block_id.substring(5);
     movesCorrector.update_undo_flip(block_id);
     flipBlock(block_id, blocks.get_block_text(id),
               blocks.get_block_color(id), currentConfig);
     movesCorrector.enable_incorrect_button();
-});
+
+};
 
 socket.on('indicate_impossible_move', function(move) {
-    let color = move['predicted_color'];
-    let letter = move['predicted_letter'];
-    let message = 'I think that this is an impossible move. '
-    + 'My prediction is that you are trying to move the '
-    + color + ' '
-    + letter + '. '
-    + 'Please check that this move is possible.';
+    let color = move['predicted_color'] || 'Ambiguous';
+    let letter = move['predicted_letter'] || 'Ambiguous';
+    let message = 'I think that this is an impossible move.\n'
+        + 'Predicted color: ' + color + '\n'
+        + 'Predicted letter: ' + letter;
+
     alert(message);
 });
 
 socket.on('indicate_ambiguous_move', function(move) {
     let color = move['predicted_color'] || 'Ambiguous';
     let letter = move['predicted_letter'] || 'Ambiguous';
     let message = 'I think that this is an ambiguous move. '
-    + 'Please try a more specific instruction.\n'
-    + 'Predicted color: ' + color + '\n'
-    + 'Predicted letter: ' + letter;
+        + 'Please try a more specific instruction.\n'
+        + 'Predicted color: ' + color + '\n'
+        + 'Predicted letter: ' + letter + '\n'
+        + 'Enter the identifier of the intended block.';
```

Comment:
Should I enter an identifier or a more specific instruction? One of those two statements should go.